src/task_monitor/tasks/insert_identities.rs

```diff
@@ -128,29 +124,14 @@ async fn insert_identities(
             "Length mismatch when appending identities to tree"
         );
 
-        let items = three_way_zip(
-            data.into_iter(),
-            identities.into_iter(),
-            on_completes.into_iter(),
-        );
+        let items = data.into_iter().zip(identities.into_iter());
 
-        for ((root, proof, leaf_index), identity, on_complete) in items {
+        for ((root, proof, leaf_index), identity) in items {
             database
                 .insert_pending_identity(leaf_index, &identity, &root)
                 .await?;
 
-            let inclusion_proof = InclusionProof {
-                status: Status::Pending,
-                root,
-                proof,
-            };
-
-            if on_complete
-                .send(OnInsertComplete::Proof(inclusion_proof))
-                .is_err()
-            {
-                error!("On complete channel was closed before identity was inserted");
-            }
+            // TODO: update db, set status of unprocessed to something?
```

Comment:
When processing finished without errors, we can safely remove this identity from `unprocessed_identities`